src/ripple/basics/strHex.h

```diff
@@ -62,22 +66,33 @@ charUnHex (char c)
 }
 /** @} */
 
-// NIKB TODO cleanup this function and reduce the need for the many overloads
-//           it has in various places.
-template<class FwdIt>
-std::string strHex (FwdIt first, int size)
+template<class InputIterator>
+std::string strHex(InputIterator begin, InputIterator end)
+{
+    std::string result;
+    result.reserve (std::distance(begin, end));
+    boost::algorithm::hex (begin, end, std::back_inserter(result));
+    return result;
+}
+template <class T>
+std::string strHex(T const& blob)
 {
-    std::string s;
-    s.resize (size * 2);
-    for (int i = 0; i < size; i++)
-    {
-        unsigned char c = *first++;
-        s[i * 2]     = charHex (c >> 4);
-        s[i * 2 + 1] = charHex (c & 15);
-    }
-    return s;
+    return strHex(blob.cbegin(), blob.cend());
 }
 
+inline std::string strHex (std::string const& src)
```

Comment:
Why do we need this overload? Doesn't the overload above do the same thing?

Reply:
The overload taking two iterators can handle a `std::string` as `std::string` has begin and end iterators, yeah. I can change all of the call sites using this explicit `std::string` overload to now have the callers use the two-iterator overload. 👍